src/BaseStrategy.sol

```diff
@@ -115,7 +117,8 @@ abstract contract BaseTokenizedStrategy {
     ITokenizedStrategy internal immutable TokenizedStrategy;
 
     // Underlying asset the Strategy is earning yield on.
-    address public immutable asset;
+    // Stored here for cheap retrievals wihtin the strategy.
+    ERC20 internal immutable asset;
```

Comment:
Curious why we decided to switch to ERC20 here. Do we always cast to ERC20?

Reply:
Not always but seems to be the majority. And this way it matches `want` from V2.